doc/source/data/getting-started.rst

```diff
@@ -1,229 +1,86 @@
 
-The batch format can be specified using ``batch_format`` option, which defaults to "native",
-meaning pandas format for Arrow-compatible batches, and Python lists for other types. You
-can also specify explicitly "arrow" or "pandas" to force a conversion to that batch format.
-The batch size can also be chosen. If not given, the batch size will default to entire blocks.
+.. literalinclude:: ./doc_code/quick_start.py
+   :language: python
+   :start-after: __data_transform_begin__
+   :end-before: __data_transform_end__
 
 .. tip::
 
-    Datasets also provides the convenience methods ``map``, ``flat_map``, and ``filter``, which are not vectorized (slower than ``map_batches``), but may be useful for development.
-
-By default, transformations are executed using Ray tasks.
-For transformations that require setup, specify ``compute=ray.data.ActorPoolStrategy(min, max)`` and Ray will use an autoscaling actor pool of ``min`` to ``max`` actors to execute your transforms.
-For a fixed-size actor pool, specify ``ActorPoolStrategy(n, n)``.
-The following is an end-to-end example of reading, transforming, and saving batch inference results using Ray Data:
-
-.. code-block:: python
-
-    from ray.data import ActorPoolStrategy
-
-    # Example of GPU batch inference on an ImageNet model.
-    def preprocess(images: List[bytes]) -> List[bytes]:
-        return images
-
-    class BatchInferModel:
-        def __init__(self):
-            self.model = ImageNetModel()
-        def __call__(self, batch: pd.DataFrame) -> pd.DataFrame:
-            return self.model(batch)
+    Datasets also provides the convenience transformation methods ``map``, ``flat_map``, and ``filter``, which are not vectorized (slower than ``map_batches``), but may be useful for development.
 
-    ds = ray.data.read_binary_files("s3://bucket/image-dir")
+The transformation is composable. You can further apply transformations on the output Dataset, forming
+a chain of transformations. Ray Dataset will optimize the execution by fusing transformations when appropriate.
 
-    # Preprocess the data.
-    ds = ds.map_batches(preprocess)
-    # -> Map Progress: 100%|████████████████████| 200/200 [00:00<00:00, 1123.54it/s]
-
-    # Apply GPU batch inference with actors, and assign each actor a GPU using
-    # ``num_gpus=1`` (any Ray remote decorator argument can be used here).
-    ds = ds.map_batches(
-        BatchInferModel, compute=ActorPoolStrategy(10, 20),
-        batch_size=256, num_gpus=1)
-    # -> Map Progress (16 actors 4 pending): 100%|██████| 200/200 [00:07, 27.60it/s]
-
-    # Save the results.
-    ds.repartition(1).write_json("s3://bucket/inference-results")
+By default, transformations are executed using Ray tasks.
+For transformations that require setup, you may want to use Ray actors by specifying ``compute=ray.data.ActorPoolStrategy(min, max)`` and Ray will use an autoscaling actor pool of ``min`` to ``max`` actors to execute your transforms.
```

Comment:
We should mention that the actor compute strategy will cache that stateful setup at actor creation time, and that's why it's useful for transformations requiring stateful setup.